### DISCUSSION TITLE
Quantum solver for differential equations

### DISCUSSION THREAD

**Discussion by ljubabu** [Qiskit/.github#13] [Projects]
Hi,

I would like to start the conversation about the potential methods for solving the linear and non-linear differential equations on quantum devices. I have some ideas for the project in which several different methodologies can be applied and tested using the Qiskit. If anyone is interested in this kind of projects, please share your thoughts.  

Ljubomir

**Comment by olgOk** [comment]
Hello! Could you please share the status of this project? Was it implemented in Qiskit (I could not find any DE solvers. Please point me out if I overlooked it). If it is still in the plans, I would be interested in joining the project.

**Reply by afrah** [reply]
I am also interested

**Comment by hrahman12** [comment]
Please see the documentation for “Solvers”. It covers how to solve linear and ordinary DE’s through classes and functions  https://qiskit.org/ecosystem/dynamics/apidocs/solvers.html

**Reply by olgOk** [reply]
Correct me if I am wrong, but these are classical DE solvers, right?

**Reply by hrahman12** [reply]
It can be both. You can model Dyson perturbation models for time dependent DE. https://qiskit.org/ecosystem/dynamics/discussions/dyson_magnus.html

https://www.researchgate.net/publication/369398851_Time-marching_based_quantum_solvers_for_time-dependent_linear_differential_equations

**Reply by olgOk** [reply]
sorry, where can I possibly find an implementation of the QSVT/QSP, matrix/scalar block encodings in Qiskit as described  in https://www.researchgate.net/publication/369398851_Time-marching_based_quantum_solvers_for_time-dependent_linear_differential_equations ?

**Reply by hrahman12** [reply]
Search with Quantum Singular Value Transformation in Qiskit https://github.com/bartubisgin/QSVTinQiskit-2021-Europe-Hackathon-Winning-Project-
